File: aquasense-backend/routes/alerts.py
"""Alertes anomalies."""

from __future__ import annotations

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from database import Alert, get_db

router = APIRouter(prefix="/alerts", tags=["alerts"])


@router.get("")
def list_alerts(user_id: int, db: Session = Depends(get_db)):
    rows = (
        db.query(Alert)
        .filter(Alert.user_id == user_id)
        .order_by(Alert.created_at.desc())
        .limit(200)
        .all()
    )
    return [
        {
            "id": a.id,
            "anomaly_type": a.anomaly_type,
            "anomaly_name": a.anomaly_name,
            "message": a.message or "",
            "confidence": float(a.confidence or 0),
            "timestamp": a.created_at.isoformat() + "Z",
            "resolved": bool(a.resolved),
            "consumption_m3": float(a.consumption_m3) if a.consumption_m3 is not None else None,
        }
        for a in rows
    ]


@router.patch("/{alert_id}/resolve")
def resolve_alert(alert_id: int, db: Session = Depends(get_db)):
    a = db.query(Alert).filter(Alert.id == alert_id).first()
    if not a:
        raise HTTPException(status_code=404, detail="Alerte introuvable")
    a.resolved = True
    db.commit()
    return {"message": "Alerte résolue"}


@router.delete("/{alert_id}")
def delete_alert(alert_id: int, db: Session = Depends(get_db)):
    a = db.query(Alert).filter(Alert.id == alert_id).first()
    if not a:
        raise HTTPException(status_code=404, detail="Alerte introuvable")
    db.delete(a)
    db.commit()
    return {"message": "Alerte supprimée"}

File: src/api/alerts.ts
import { api } from './client';

export type AlertDto = {
  id: number;
  anomaly_type: number;
  anomaly_name: string;
  message: string;
  confidence: number;
  timestamp: string;
  resolved: boolean;
  consumption_m3: number | null;
};

export async function getAlerts(user_id: number): Promise<AlertDto[]> {
  const { data } = await api.get<AlertDto[]>('/alerts', { params: { user_id } });
  return data;
}

export async function resolveAlert(id: number): Promise<{ message: string }> {
  const { data } = await api.patch<{ message: string }>(`/alerts/${id}/resolve`);
  return data;
}

export async function deleteAlert(id: number): Promise<{ message: string }> {
  const { data } = await api.delete<{ message: string }>(`/alerts/${id}`);
  return data;
}
